tests/test_enterprise/api/test_views.py

```diff
@@ -2890,84 +2889,68 @@ def test_enterprise_customer_catalogs_response_formats(self):
-            settings.TEST_SERVER + reverse(self.CATALOG_QUERY_ENDPOINT, kwargs={'catalog_query_id': 1}),
-            data=json.dumps({'current_troll_hunter': 'Jim Lake Jr.'}),
-            content_type='application/json'
-        )
-        assert response.status_code == 405
-        response = response.json()
-        assert response['detail'] == 'Method "POST" not allowed.'
+        ENTERPRISE_CATALOG_QUERY_ENDPOINT = reverse('enterprise_catalog_query-list')
 
-    def test_get_catalog_query_not_staff(self):
+        response = self.client.get(ENTERPRISE_CATALOG_QUERY_ENDPOINT)
+        self.assertEqual(response.status_code, 200)
+
+    def test_enterprise_catalog_query_detail(self):
         """
-        Test that `CatalogQueryView` does not allow non staff users.
+        ``enterprise_catalog_query``'s response when a catalog uuid is provided.
         """
-        # Creating a non staff user so as to verify the insufficient permission conditions.
-        user = factories.UserFactory(username='test_user', is_active=True, is_staff=False)
-        user.set_password('test_password')
-        user.save()
+        ENTERPRISE_CATALOG_QUERY_ENDPOINT = reverse('enterprise_catalog_query-detail', args=[1])
 
-        client = APIClient()
-        client.login(username='test_user', password='test_password')
-        response = client.get(
-            settings.TEST_SERVER + reverse(self.CATALOG_QUERY_ENDPOINT, kwargs={'catalog_query_id': 1})
-        )
+        response = self.client.get(ENTERPRISE_CATALOG_QUERY_ENDPOINT)
+        self.assertEqual(response.status_code, 200)
 
-        assert response.status_code == 403
-        response = response.json()
-        assert response['detail'] == 'You do not have permission to perform this action.'
+    def test_enterprise_catalog_query_detail_not_found(self):
+        """
+        ``enterprise_catalog_query``'s response when a catalog uuid is provided but not found.
+        """
+        ENTERPRISE_CATALOG_QUERY_ENDPOINT = reverse('enterprise_catalog_query-detail', args=[2])
 
-    def test_get_catalog_query_not_logged_in(self):
+        response = self.client.get(ENTERPRISE_CATALOG_QUERY_ENDPOINT)
+        self.assertEqual(response.status_code, 404)
+
+    def test_enterprise_catalog_query_detail_bad_uuid(self):
         """
-        Test that `CatalogQueryView` only allows logged in users.
+        ``enterprise_catalog_query``'s response when a catalog uuid is provided but bad.
         """
-        client = APIClient()
-        # User is not logged in.
-        response = client.get(
-            settings.TEST_SERVER + reverse(self.CATALOG_QUERY_ENDPOINT, kwargs={'catalog_query_id': 1})
-        )
-        assert response.status_code == 403
-        response = response.json()
-        assert response['detail'] == 'Authentication credentials were not provided.'
+        ENTERPRISE_CATALOG_QUERY_ENDPOINT = reverse('enterprise_catalog_query-detail', args=['bad-uuid'])
```

Comment:
shouldn't 'bad-uuid' result in a 400 error to indicate that uuid validation failed?